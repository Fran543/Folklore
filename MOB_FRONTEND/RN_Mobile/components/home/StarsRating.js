import { StyleSheet } from 'react-native';
import { Rating } from 'react-native-ratings';
import { useEffect, useState } from 'react';
import EndPoints from '../../constants/endPoints';

export default function StarsRating(props) {
    const [orgScore, setOrgScore] = useState(0)

    useEffect(() => {
        fetch(EndPoints.getUserStoryScore, {
            method: 'POST',
            credentials: 'include',
            headers: { 'Content-Type': 'application/json' },
            body: JSON.stringify({ idStory: props.idStory, userID: props.idUser })
        })
            .then(res => res.json())
            .then(
                (result) => {
                    result.score && setOrgScore(result.score)
                    console.log(result.score)
                },
                (error) => {
                    console.log(error)
                }
            )
    }, [])

    function saveScore(score) {
        fetch(EndPoints.addScoreToStory, {
            method: 'POST',
            credentials: 'include',
            headers: { 'Content-Type': 'application/json' },
            body: JSON.stringify({ score: score, idStory: props.idStory })
        })
            .then(res => res.json())
            .then(
                (result) => {
                    console.log(result)
                },
                (error) => {
                    console.log(error)
                }
            )
    }

    return (
        <Rating
            type='star'
            ratingColor='black'
            ratingBackgroundColor='#BDC3C7'
            ratingCount={5}
            startingValue={orgScore}
            imageSize={50}
            tintColor='#333333'
            style={styles.starContainer}
            onFinishRating={saveScore}
        />
    );
}

const styles = StyleSheet.create({
    starContainer: {
        paddingVertical: 20,
    }
})