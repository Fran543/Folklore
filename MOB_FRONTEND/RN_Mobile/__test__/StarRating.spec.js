import React from 'react';
import renderer from 'react-test-renderer';

import StarsRating from '../components/home/StarsRating';

describe('<StarsRating />', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<StarsRating idStory="1"/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('has 1 child', () => {
        const tree = renderer.create(<StarsRating idStory="1"/>).toJSON();
        expect(tree.children.length).toBe(1);
    });
});
