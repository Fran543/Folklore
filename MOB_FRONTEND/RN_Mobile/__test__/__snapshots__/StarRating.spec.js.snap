// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StarsRating /> renders correctly 1`] = `
<View
  pointerEvents="auto"
  style={
    Object {
      "paddingVertical": 20,
    }
  }
>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "alignItems": "center",
          "bottom": 0,
          "flexDirection": "row",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "backgroundColor": "#f1c40f",
            "height": 50,
            "width": 0,
          }
        }
      />
      <View
        collapsable={false}
        style={
          Object {
            "backgroundColor": "white",
            "height": 50,
            "width": 250,
          }
        }
      />
    </View>
    <View>
      <Image
        source={1}
        style={
          Object {
            "height": 50,
            "tintColor": "#333333",
            "width": 50,
          }
        }
      />
    </View>
    <View>
      <Image
        source={1}
        style={
          Object {
            "height": 50,
            "tintColor": "#333333",
            "width": 50,
          }
        }
      />
    </View>
    <View>
      <Image
        source={1}
        style={
          Object {
            "height": 50,
            "tintColor": "#333333",
            "width": 50,
          }
        }
      />
    </View>
    <View>
      <Image
        source={1}
        style={
          Object {
            "height": 50,
            "tintColor": "#333333",
            "width": 50,
          }
        }
      />
    </View>
    <View>
      <Image
        source={1}
        style={
          Object {
            "height": 50,
            "tintColor": "#333333",
            "width": 50,
          }
        }
      />
    </View>
  </View>
</View>
`;
